web/jobs/jobs.go

```diff
@@ -466,6 +467,94 @@ func cleanJobs(c echo.Context) error {
 	return c.JSON(200, map[string]int{"deleted": len(ups)})
 }
 
+func purgeJobs(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	if err := middlewares.AllowWholeType(c, webpermissions.DELETE, consts.Jobs); err != nil {
+		return err
+	}
+
+	workersParam := c.QueryParam("workers")
+	monthsParam := c.QueryParam("months")
+
+	var months int = 2
+	var err error
+	if monthsParam != "" {
+		months, err = strconv.Atoi(monthsParam)
+		if err != nil {
+			return err
+		}
+	}
+	workers := jobs.GetWorkersNamesList()
+	if workersParam != "" {
+		workers = strings.Split(workersParam, ",")
+	}
+
+	// Step 1: We want to get all the jobs prior to the date parameter.
+	// Jobs returned are the ones we want to remove
+	d := time.Now().AddDate(0, -months, 0)
+	jobsBeforeDate, err := jobs.GetJobsBeforeDate(instance, d)
+	if err != nil && err != jobs.ErrNotFoundJob {
+		return err
+	}
+	// Step 2: We also want to keep a minimum number of jobs for each state.
+	// Jobs returned will be kept.
+	limits := map[jobs.State]int{
+		jobs.Queued:  0, // "0" is equivalent of the default limit
+		jobs.Running: 0,
+		jobs.Done:    0,
+		jobs.Errored: 0,
+	}
+
+	var lastsJobs []*jobs.Job
```

Comment:
I'd have use a map of id for `lastJobs` (`map[string]struct{}{}`) to make it easier to look if a job id is one of the last jobs or not.